### PR TITLE
Added configuration for Embarcadero-Borland Clang Compiler 64 and rename BC-32 to BCC-32

### PR DESCRIPTION
**Pull request by GermanAizek**

### CODE REVIEW

File: Configurations/50-cppbuilder.conf [hunk 1]
```diff
@@ -58,5 +58,65 @@ my %targets = (
...
+        inherit_from     => [ "BASE_Windows" ],
+        sys_id           => "WIN64",
+        bn_ops           => "BN_LLONG",
+        thread_scheme    => "winthreads",
+        cc               => "bcc64",
+        CPP              => "cpp64 -oCON -Sc -Sr",
+        defines          => add("WIN64_LEAN_AND_MEAN", "OPENSSL_SYS_WIN64",
+                                "L_ENDIAN", "DSO_WIN64", "_stricmp=stricmp",
+                                "_strnicmp=strnicmp", "_malloca=malloc",
+                                "_freea=free", "_setmode=setmode"),
```
Comment (t8m) [inline Configurations/50-cppbuilder.conf:72]: Why are the _malloca and _freea aliases needed? I hope we are not using these functions anywhere in the code.

File: Configurations/50-cppbuilder.conf [hunk 2]
```diff
@@ -1,5 +1,5 @@
 my %targets = (
-    "BC-32" => {
+    "BCC-32" => {
```
Comment (t8m) [inline Configurations/50-cppbuilder.conf:2]: I would not rename it, or rather make the old name be an alias of the new one.

**Review by t8m — CHANGES_REQUESTED** [review]

**Comment by t8m** [comment]
We will need a CLA to be signed to be able to accept this contribution. Could you please follow: https://www.openssl.org/policies/cla.html

**Comment by GermanAizek** [comment]
Close PR and migrate to here: https://github.com/openssl/openssl/pull/23377

**Pull request closed without merging**